chain/deals/client_states.go

```diff
@@ -126,16 +133,72 @@ func (c *Client) staged(ctx context.Context, deal ClientDeal) error {
 }
 
 func (c *Client) sealing(ctx context.Context, deal ClientDeal) error {
-	resp, err := c.readStorageDealResp(deal)
-	if err != nil {
-		return err
+
+	// TODO: disconnect
+
+	checkFunc := func(ts *types.TipSet) (done bool, more bool, err error) {
+		sd, err := stmgr.GetStorageDeal(ctx, c.sm, deal.DealID, ts)
+		if err != nil {
+			// TODO: This may be fine for some errors
+			return false, false, xerrors.Errorf("failed to look up deal on chain: %w", err)
+		}
+
+		if sd.ActivationEpoch > 0 {
+			select {
+			case c.updated <- clientDealUpdate{
+				newState: api.DealComplete,
+				id:       deal.ProposalCid,
+			}:
+			case <-c.stop:
+			}
+
+			return true, false, nil
+		}
+
+		return false, true, nil
+	}
+
+	called := func(msg *types.Message, ts *types.TipSet, curH uint64) (bool, error) {
+		// TODO: handle errors
+
+		if msg == nil {
+			log.Error("timed out waiting for deal activation... what now?")
+			return false, nil
+		}
+
+		// TODO: can check msg.Params to see if we should even bother checking the state
+
+		sd, err := stmgr.GetStorageDeal(ctx, c.sm, deal.DealID, ts)
+		if err != nil {
+			return false, xerrors.Errorf("failed to look up deal on chain: %w", err)
+		}
+
+		if sd.ActivationEpoch == 0 {
+			return true, nil
+		}
+
+		log.Info("Storage deal %d activated at epoch %d", deal.DealID, sd.ActivationEpoch)
+
+		select {
+		case c.updated <- clientDealUpdate{
+			newState: api.DealComplete,
+			id:       deal.ProposalCid,
+		}:
+		case <-c.stop:
+		}
+
+		return false, nil
 	}
 
-	if resp.State != api.DealComplete {
-		return xerrors.Errorf("deal wasn't complete (State=%d)", resp.State)
+	revert := func(ctx context.Context, ts *types.TipSet) error {
+		log.Warn("deal activation reverted; TODO: actually handle this!")
+		// TODO: Just go back to DealSealing?
```

Comment:
Probably do that (check how that affects event handling when we return `false` for `more`)

Reply:
(Can fix later)